src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx

```diff
@@ -0,0 +1,307 @@
+  TwitterLogo,
+  TelegramLogo,
+  GithubLogo,
+  FacebookLogo,
+  InstagramLogo,
+} from 'phosphor-react';
+import clsx from 'clsx';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  FieldValues,
+  SubmitHandler,
+  useController,
+  useForm,
+  useWatch,
+} from 'react-hook-form';
+import { createColumnHelper, ColumnDef } from '@tanstack/react-table';
+import { useActionSidebarContext } from '~context/ActionSidebarContext';
+import { LINK_TYPE, LinkType, SocialLinksTableModel } from './types';
+import { formatText } from '~utils/intl';
+import { TableWithMeatballMenuProps } from '~v5/common/TableWithMeatballMenu/types';
+import RowField from './partials/RowField';
+
+export const useSocialLinksTableColumns = (
+  name: string,
+): ColumnDef<SocialLinksTableModel, string>[] => {
+  const columnHelper = useMemo(
+    () => createColumnHelper<SocialLinksTableModel>(),
+    [],
+  );
+
+  const columns: ColumnDef<SocialLinksTableModel, string>[] = useMemo(
+    () => [
+      columnHelper.display({
+        id: 'name',
+        header: () => formatText({ id: 'table.row.type' }),
+        cell: ({ row }) => (
+          <RowField key={row.id} name={`${name}.${row.index}.name`} />
+        ),
+      }),
+      columnHelper.display({
+        id: 'link',
+        header: () => formatText({ id: 'table.row.url' }),
+        cell: ({ row }) => (
+          <RowField key={row.id} name={`${name}.${row.index}.link`} />
+        ),
+      }),
+    ],
+    [columnHelper, name],
+  );
+
+  return columns;
+};
+
+export const useGetTableMenuProps = ({ remove }, data) => {
+  const {
+    socialLinksModalToggle: [, { toggleOn: toggleChangeSocialLinksModalOn }],
+  } = useActionSidebarContext();
+  const { field } = useController({
+    name: 'name',
+  });
+
+  return useCallback<
+    TableWithMeatballMenuProps<SocialLinksTableModel>['getMenuProps']
+  >(
+    ({ index }) => ({
+      cardClassName: 'min-w-[9.625rem] whitespace-nowrap',
+      items: [
+        {
+          key: 'edit',
+          onClick: () => {
+            field.onChange({ name: data[index].name, link: data[index].link });
+            toggleChangeSocialLinksModalOn();
+          },
+          label: formatText({ id: 'table.row.edit.link' }),
+          iconName: 'edit-pencil',
+        },
+        {
+          key: 'remove',
+          onClick: () => remove(index),
+          label: formatText({ id: 'table.row.remove' }),
+          iconName: 'trash',
+        },
+      ],
+    }),
+    [data, remove, field, toggleChangeSocialLinksModalOn],
+  );
+};
+
+export const useSocialLinksButtons = (
+  selectedLink: LinkType,
+): {
+  type: LinkType;
+  icon: JSX.Element;
+}[] => {
+  const socialLinksButtons = [
+    {
+      type: LINK_TYPE.Custom,
+      icon: (
+        <span className="mr-2">
+          <GlobeHemisphereEast size={16} />
+        </span>
+      ),
+    },
+    {
+      type: LINK_TYPE.Whitepaper,
+      icon: (
+        <span
```

Comment:
There's a lot of code duplication. Maybe you can create a component - list - and just pass what you need there?